explainer.md

```diff
@@ -0,0 +1,259 @@
+# Introduction
+When a user navigates on the web, they get to see the inner workings of web experiences: flash of white followed by a piece-meal rendering phase. This sequenced user experience results in a higher cognitive load because the user has to connect the dots between where they were, and where they are. Not only that, a smooth animation to transition between scenes also reduces the loading latency perceived by the user even if the actual loading time is the same. For these reasons, most platforms provide easy to use primitives to enable developers to build seamless transitions: [Android](https://developer.android.com/training/transitions/start-activity), [iOS/Mac](https://developer.apple.com/documentation/uikit/uimodaltransitionstyle) and [Windows](https://docs.microsoft.com/en-us/windows/apps/design/motion/page-transitions).
+
+This feature provides developers with the same capability on the web, irrespective of whether the scenes in a transition are rendered across Documents.
+
+# Use-Cases
+A visual demo of the transition patterns targeted by this feature are [here](https://material.io/design/motion/the-motion-system.html#transition-patterns). The following is a summary of the semantics of these transition patterns :
+
+* Root Transitions : The full page content animates between 2 web pages with an optional static UI element on top. [Shared axis](https://material.io/design/motion/the-motion-system.html#shared-axis) shows an example.
+* Shared Element Transitions : A persistent UI element morphs into another (which could be a UI element on the next page or the whole page) changing its shape and content. [Container transform](https://material.io/design/motion/the-motion-system.html#container-transform) shows an example.
+* Entry/Exit Transitions : A UI element animates as it exits or enters the screen. This [issue](https://github.com/WICG/shared-element-transitions/issues/37) shows an example.
+
+These transitions should be feasible in SPAs (Single Page Apps) and MPAs (Multi Page Apps).
+
+# Glossary
```

Comment:
Rather than a long list of glossary and contents at the top, let's go straight to an example. I propose you give an example of navigating from one page that is a solid color with a single rectangle shared element on it, to a page that is some other solid color with the same shared element at a different position.

Old page:
<html style="background: blue">
  <div sharedId="header" style="width: 50px; height: 50px; background: green; position: relative"></div>
</html>


New page:
  <div sharedId="header" style="width: 50px; height: 50px; background: green; position: relative; top: 300px"></div>
</html>

The API would be something like:

Old document:
addEventListener("navigate", (event) => {
  document.offerTransition({sharedElements: ["header"]});
}

New one:
<html transition="fadeIn" transitionDuration=0.1s>
...


Then you can proceed directly to the steps:

1. When offerTransition is called:
1a. Save the output of element(sharedId) for each shared id offered.
1b. Save the output of element(html).
2. Navigate to the new page, while leaving the last rendered pixels of the old page on the screen.
3. If the new page has a "transition" attribute on the HTML element, suppress rendering until DOMContentLoaded is completed.
4. After DOMContentLoaded has fired, start an animation as follows:
4a. Create a container element named and its child in the top layer for each shared element |name| in the current document. The children are accessible via a pseudo-element selector ::new-shared-element(|name|)
4b. Create a container element and child for each shared element in the old document, and identify via ::old-shared-element(|name|)
4b. Apply a UA style sheet with ... in it. (element() function, opacity:0 on shared elements in the page, etc
4c. Render as normal
5. At the completion of the animation time specified in the transitionDuration attribute, remove all pseudo-elements from the top layer.

Then say that the following changes to the element() spec would be made:

1. containment
2. no breaking
3. no visual effects
4. element(thing in old document) refers to a snapshot from 1a/1b

If you say all this in the language of the CSSWG specs, you don't need a lot of the explanatory text about what "painted content" means.

Reply:
One more thing, the no visual effects modification seems difficult to explain in a way that would be backwards compatible. And it also opens the discussion for why only those styles and not box decorations. I feel like if we're suggesting that, we might as well be discussing the "Natively Supported Live Animatable Properties" alternative.